cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -0,0 +1,346 @@
+use core::fmt;
+use std::fs::read_to_string;
+use std::path::Path;
+use std::{env, fs, io};
+
+use clap::Parser;
+use std::num::NonZeroU32;
+use std::sync::atomic::AtomicBool;
+use toml_edit::{Document, Formatted, InlineTable, TomlError, Value};
+
+#[derive(Clone, Debug, PartialEq, clap::ValueEnum)]
+pub enum ExampleContract {
+    Account,
+    Alloc,
+    AtomicMultiswap,
+    AtomicSwap,
+    Auth,
+    CrossContract,
+    CustomTypes,
+    DeepContractAuth,
+    Deployer,
+    Errors,
+    Events,
+    Fuzzing,
+    HelloWorld,
+    Increment,
+    LiquidityPool,
+    Logging,
+    SimpleAccount,
+    SingleOffer,
+    Timelock,
+    Token,
+    UpgradeableContract,
+    None,
+}
+
+impl fmt::Display for ExampleContract {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ExampleContract::Account => write!(f, "account"),
+            ExampleContract::Alloc => write!(f, "alloc"),
+            ExampleContract::AtomicMultiswap => write!(f, "atomic_multiswap"),
+            ExampleContract::AtomicSwap => write!(f, "atomic_swap"),
+            ExampleContract::Auth => write!(f, "auth"),
+            ExampleContract::CrossContract => write!(f, "cross_contract"),
+            ExampleContract::CustomTypes => write!(f, "custom_types"),
+            ExampleContract::DeepContractAuth => write!(f, "deep_contract_auth"),
+            ExampleContract::Deployer => write!(f, "deployer"),
+            ExampleContract::Errors => write!(f, "errors"),
+            ExampleContract::Events => write!(f, "events"),
+            ExampleContract::Fuzzing => write!(f, "fuzzing"),
+            ExampleContract::HelloWorld => write!(f, "hello_world"),
+            ExampleContract::Increment => write!(f, "increment"),
+            ExampleContract::LiquidityPool => write!(f, "liquidity_pool"),
+            ExampleContract::Logging => write!(f, "logging"),
+            ExampleContract::SimpleAccount => write!(f, "simple_account"),
+            ExampleContract::SingleOffer => write!(f, "single_offer"),
+            ExampleContract::Timelock => write!(f, "timelock"),
+            ExampleContract::Token => write!(f, "token"),
+            ExampleContract::UpgradeableContract => write!(f, "upgradeable_contract"),
+            ExampleContract::None => write!(f, "none"),
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    pub project_path: String,
+
+    /// optional flag to specify soroban example contracts to include
+    #[arg(short, long, num_args = 1..=20, default_value = "none")]
```

Comment:
`increment` gets installed no matter what, currently. If I omit all `-w` options, it gets installed. If I only say `-w hello-world`, both `increment` and `hello-world` get added. If I say `-w increment`, then I get this output:

```
cargo run -- contract init ../lol -w increment
    Finished dev [unoptimized + debuginfo] target(s) in 0.27s
     Running `target/debug/soroban contract init ../lol -w increment -w hello-world`
ℹ️  Initializing project at ../lol
➕  Writing ../lol/Cargo.toml
➕  Writing ../lol/contracts/increment/Cargo.toml
➕  Writing ../lol/contracts/increment/src/test.rs
➕  Writing ../lol/contracts/increment/src/lib.rs
➕  Writing ../lol/README.md
➕  Writing ../lol/.gitignore
ℹ️  Initializing example contract: increment
ℹ️  Skipped creating ../lol/contracts/increment/Cargo.toml as it already exists
ℹ️  Skipped creating ../lol/contracts/increment/src/test.rs as it already exists
ℹ️  Skipped creating ../lol/contracts/increment/src/lib.rs as it already exists

Reply:
Ah, nice catch! In some of the convo above on this PR we had talked about always including the increment contract by default so that the user would have an example to go off. But I can rework this interaction, so it's a bit more intuitive.

Reply:
Could we make the default contract that is written out a new contract template rather than one of the examples? It could contain the same logic as increment, but named 'contract/' so it is more a boilerplate for someone to work in, rather than an example named something else.

Reply:
Just pushed up a change in [4f7eac4](https://github.com/stellar/soroban-tools/pull/1156/commits/4f7eac47456d8724a9f0ce86770c4b34415eac83) and [ffc4d80](https://github.com/stellar/soroban-tools/pull/1156/commits/ffc4d80f2d12198fde747995ee64116ffb11a498) that will include hello-world by default, and removes it from the list of options so no longer have that confusing `Skipped creating...` for it.